rag/svr/task_executor.py

```diff
@@ -735,10 +739,11 @@ async def main():
     async with trio.open_nursery() as nursery:
         nursery.start_soon(report_status)
         while not stop_event.is_set():
-            async with task_limiter:
-                nursery.start_soon(handle_task)
+            nursery.start_soon(limited_handle_task)
+            await trio.sleep(1)
```

Comment:
Good catch! However your change is smell. It creates an async task every 1 second so there may be very many `limited_handle_task` tasks while only 5 of them are inside `handle_task`.